src/Components/MainPage.js

```diff
@@ -18,9 +27,12 @@ export default class MainPage extends React.Component {
 
                 <Col sm={12} md={3}>
                     {/* <div className="sample-fill"/> */}
-                    <CourseCart> </CourseCart>
+                    <CourseCart showModal={this.showModal} hideModal={this.hideModal}> </CourseCart>

```

Comment:
I'm not sure what you mean by this. I have the state of the modal being kept in MainPage.js and then i'm just passing the state and show/hide functions to the other components.

Reply:
showModal, and hideModal are just the passed functions from MainPage.js, they aren't the state of the modal.

Reply:
Yes, correct. 
I meant, instead of passing a callback function to update state, it should be updated solely from MainPage.js

In other words, the modal should have no idea that it is being opened and closed. 
The code I shared above should do the trick for that